### DISCUSSION TITLE
Performance vs NuGet server

### DISCUSSION THREAD

**Discussion by Mangopony** [loic-sharma/BaGet#756] [General]
Hi guys,

Has anyone tried a like for like performance comparison vs Microsoft NuGet server running on-premise?
I'm currently using NuGet and it's not very scalable, it just gets slower and slower as the number of packages keeps increasing.
It can take more than 20 seconds to figure out the dependencies when restoring the packages.

Thank you
Ion

**Comment by loic-sharma** [comment]
I would expect BaGet to scale better than NuGet.Server. BaGet's original goal was to [replace nuget.org's implementation and to scale to >1K reads per second](https://github.com/loic-sharma/BaGet/issues/71). Back then BaGet could ingest millions of packages from nuget.org in under a day.

You can test performance yourself by importing your packages from NuGet.Server: https://loic-sharma.github.io/BaGet/import/nugetserver/

**Comment by Mangopony** [comment]
Thank you Loic, I intend to give it a try. I will let you know how it goes.
CheersIon
    On Friday, December 30, 2022 at 07:50:15 PM GMT, Loïc Sharma ***@***.***> wrote:  
 
 


I would expect BaGet to scale better than NuGet.Server. BaGet's original goal was to replace nuget.org's implementation and to scale to >1K reads per second. Back then BaGet could ingest millions of packages from nuget.org in under a day.

You can test performance yourself by importing your packages from NuGet.Server: https://loic-sharma.github.io/BaGet/import/nugetserver/
